internal/exp/metrics/identity/metric.go

```diff
@@ -0,0 +1,65 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package identity
+
+import (
+	"hash"
+
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+)
+
+type Metric struct {
+	scope Scope
+
+	name string
+	unit string
+	ty   pmetric.MetricType
+
+	monotonic   bool
+	temporality pmetric.AggregationTemporality
+}
+
+func (i Metric) Hash() hash.Hash64 {
+	sum := i.scope.Hash()
+	sum.Write([]byte(i.name))
+	sum.Write([]byte(i.unit))
+
+	var mono byte
+	if i.monotonic {
+		mono = 1
+	}
+	sum.Write([]byte{byte(i.ty), mono, byte(i.temporality)})
+	return sum
+}
+
```

Comment:
Can we add functions for each identity struct to return the "inner" ID? For example:

```
func (i Metric) Scope() Scope {
    return i.scope
}
```

So `Stream` has a `Metric()`, `Metric` has a `Scope()`, and `Scope` has a `Resource()`.